src/libraries/Microsoft.PowerFx.Core/Public/Engine.cs

```diff
@@ -50,11 +50,21 @@ public IEnumerable<string> GetAllFunctionNames()
         /// </summary>
         /// <param name="alternateConfig">An alternate config that can be provided. Should default to engine's config if null.</param>
         /// <returns></returns>
-        private protected virtual SimpleResolver CreateResolver(PowerFxConfig alternateConfig = null)
+        private protected virtual SimpleResolver CreateSimpleResolver(PowerFxConfig alternateConfig = null)
```

Comment:
>CreateSimpleResolver [](http://example.com/codeflow?start=49&length=20)

can we avoid the dual chaining virtuals? Maybe just have CreateResolver() return INameResolver? 
I see there are a few places that ask for SimplerResolver, but they can probably just take INameResolver

Reply:
Yes, that worked out.